### PR TITLE
Changes in the source and datapack.

### PR DESCRIPTION
**Pull request by dudantas**
Synchronizations with the TFS source/pack and others changes:

* Fixed:
- [x] Bug in droploot
- [x] Bug in monsters for revscryptsys
- [x] Issue in the houses system #724
- [x] Bug at the blessings system (sometimes dropped an item from the set or lost several levels, even with bless)
- [x] Fix for: https://github.com/opentibiabr/otservbr-global/issues/963

* Changes:
- [x] Experience stages moved to Lua (credits for @ranisalt), new file stages in folder "data/stages.lua"
Note: config.lua "rateExp" doesn't change the exp anymore, it's just for fallback
Change the rate / stage experience in the "data/stages.lua" file
- [x] Moved encryption to XTEA
- [x] House functions moved back to housetile.cpp
- [x] Change the speed of all monsters in the XML script. (The speed always needs to be multiplied by the monster real speed.)

* Removed:
- [x] Exhaust for look
- [x] Exhaust for move
- [x] Exhaust for trade
- [x] Exhaust for change outfit
- [x] Exhaust for speak
- [x] Spdlog lib
- [x] Multiplication of 2.5 * for monsters speed in the source

* Added:
- [x] Bless talkaction
- [x] Sigbreak handler for windows
- [x] Revscriptys check register interface

* Added missing functions:
- [x] hasFlag (group)
- [x] getMana (player)
- [x] addMana (player)
- [x] getMaxMana (player)
- [x] hasCondition (creature)
- [x] getMonsterTypes (game)
- [x] addLoot (monsterType)
- [x] registerEvent (monsterType)

Some modifications were just to move functions to other locations, staying as in TFS and making it easier in case someone needs to make a diff between the source OTBR and TFS.

Credits to: 
The Forgotten Server
@dudantas 
@gpedro (for the fixes on the monsters XML speed.)

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -5229,69 +5168,6 @@ void Game::loadPlayersRecord()
...
-				stages[lastStageLevel] = multiplier;
-			} else {
-				for (uint32_t i = minLevel; i <= maxLevel; ++i) {
-					stages[i] = multiplier;
-				}
-			}
-		}
-	}
-	return true;
-}
```
Comment (Leo32onGIT) [inline src/game.cpp:5293]: Are you removing stages.xml?
Users existing code that is referencing stages.xml, will either be stale or broken after this.

i.e. ACC Makers do this

Comment (dudantas) [reply src/game.cpp:5293]: Ok. Another script will have to be made for the sites, it is not a difficult world to do. I'm moving the stages to Lua, as you can see here: https://github.com/opentibiabr/otservbr-global/blob/ae3946a94d280c6f9bc0e34be9600463b5cf0ddd/data/events/scripts/player.lua#L24

I did the tests and it seemed ok, better refactored and easier to change.

File: data/creaturescripts/scripts/others/droploot.lua [hunk 2]
```diff
@@ -1,10 +1,10 @@
 function onDeath(player, corpse, killer, mostDamage, unjustified, mostDamage_unjustified)
-	if getPlayerFlagValue(player, PlayerFlag_NotGenerateLoot) then --or player:getVocation():getId() == VOCATION_NONE then
+	if player:hasFlag(PlayerFlag_NotGenerateLoot) or player:getVocation():getId() == VOCATION_NONE then
```
Comment (gpedro) [inline data/creaturescripts/scripts/others/droploot.lua]: players without vocation cannot drop loot?

File: data/events/scripts/player.lua [hunk 3]
```diff
@@ -18,6 +18,32 @@ local STORAGE_CAPACITY_IMBUEMENT = 42154
 -- Players cannot throw items on teleports if set to true
 local blockTeleportTrashing = true
 
+-- Minlevel and multiplier are MANDATORY
+-- Maxlevel is OPTIONAL, but is considered infinite by default
+-- Create a stage with minlevel 1 and no maxlevel to disable stages
+	experienceStages = {
```
Comment (gpedro) [inline data/events/scripts/player.lua]: we cannot move this config to another file?

File: src/configmanager.cpp [hunk 4]
```diff
@@ -106,7 +104,7 @@ bool ConfigManager::load()
 
 	luaL_openlibs(L);
 
-	if (luaL_dofile(L, configFileLua.c_str())) {
```
Comment (gpedro) [inline src/configmanager.cpp]: i disagree with that. once we can use a custom config file using argument `-c xablau.lua`

https://github.com/opentibiabr/otservbr-global/blob/93e13351ee12661c84693f208723c9abcd44dff4/src/otserv.cpp#L139-L144

File: src/connection.cpp [hunk 5]
```diff
@@ -265,23 +263,23 @@ void Connection::parsePacket(const boost::system::error_code& error)
 				return;
 			}
 		} else {
-			msg.skipBytes(1); // Skip protocol ID
+			msg.skipBytes(1);    // Skip protocol ID
```
Comment (gpedro) [inline src/connection.cpp]: trailing spaces

File: src/connection.cpp [hunk 6]
```diff
@@ -265,23 +263,23 @@ void Connection::parsePacket(const boost::system::error_code& error)
...
 		}
 
 		protocol->onRecvFirstMessage(msg);
 	} else {
-		protocol->onRecvMessage(msg); // Send the packet to the current protocol
+		protocol->onRecvMessage(msg);    // Send the packet to the current protocol
```
Comment (gpedro) [inline src/connection.cpp]: trailing spaces

File: src/const.h [hunk 7]
```diff
@@ -256,8 +253,6 @@ enum MessageClasses : uint8_t {
 	MESSAGE_PARTY = 35, /*White message in channel (+ channelId)*/
 	MESSAGE_EVENT_ORANGE = 36, /*Orange message in the console*/
 	MESSAGE_STATUS_CONSOLE_ORANGE = 37,  /*Orange message in the console*/
-
-	MESSAGE_STATUS_CONSOLE_BLUE = MESSAGE_EVENT_ADVANCE,
```
Comment (gpedro) [inline src/const.h:260]: have you checked all references?

File: src/protocolold.cpp [hunk 8]
```diff
@@ -1,8 +1,6 @@
 /**
- * @file protocolold.cpp
```
Comment (gpedro) [inline src/protocolold.cpp:2]: protocolold is unused, maybe we can remove this protocol.

File: src/protocollogin.h [hunk 9]
```diff
@@ -1,8 +1,6 @@
 /**
- * @file protocollogin.h
```
Comment (gpedro) [inline src/protocollogin.h:2]: protocollogin is unused too

File: src/game.cpp [hunk 10]
```diff
@@ -4990,7 +4930,6 @@ void Game::shutdown()
 {
 	std::cout << "Shutting down..." << std::flush;
 
-	saveGameState();
```
Comment (gpedro) [inline src/game.cpp:4993]: we need to check if when press ctrl-c to finish the process, he still saving the server

File: data/lib/compat/compat.lua [hunk 11]
```diff
@@ -319,36 +319,36 @@ addOutfitCondition = Condition.setOutfit
...
-function getCreatureOutfit(cid) local c = Creature(cid) return c ~= nil and c:getOutfit() or false end
-function getCreatureSpeed(cid) local c = Creature(cid) return c ~= nil and c:getSpeed() or false end
-function getCreatureBaseSpeed(cid) local c = Creature(cid) return c ~= nil and c:getBaseSpeed() or false end
+function isCreature(cid) return Creature(cid) end
+function isPlayer(cid) return Player(cid) end
+function isMonster(cid) return Monster(cid) end
+function isSummon(cid) return Creature(cid):getMaster() end
+function isNpc(cid) return Npc(cid) end
+function isItem(uid) return Item(uid) end
+function isContainer(uid) return Container(uid) end
```
Comment (gpedro) [inline data/lib/compat/compat.lua]: and you double check this method? Methods with prefix `is` should returns boolean and `Creature(cid)` returns an obj

**Review by gpedro — CHANGES_REQUESTED** [review]

File: data/events/scripts/player.lua (hunk 3 above)
Comment (dudantas) [reply data/events/scripts/player.lua]: Yes we can.
I initially thought about moving to config.lua, but I ran into problems reading login.php and installing the websites.
Any suggestion?

File: src/game.cpp (hunk 10 above)
Comment (dudantas) [reply src/game.cpp:4993]: I'll check.

File: src/protocolold.cpp (hunk 8 above)
Comment (dudantas) [reply src/protocolold.cpp:2]: Ok

File: src/const.h (hunk 7 above)
Comment (dudantas) [reply src/const.h:260]: @gpedro yes, it's deprecated.

File: src/game.cpp (hunk 10 above)
Comment (dudantas) [reply src/game.cpp:4993]: @gpedro checked, working.

File: src/configmanager.cpp (hunk 4 above)
Comment (dudantas) [reply src/configmanager.cpp]: Reverted

File: src/protocollogin.h (hunk 9 above)
Comment (dudantas) [reply src/protocollogin.h:2]: Removed

File: data/lib/compat/compat.lua (hunk 11 above)
Comment (dudantas) [reply data/lib/compat/compat.lua]: Reverted to original changes.

File: data/events/scripts/player.lua (hunk 3 above)
Comment (dudantas) [reply data/events/scripts/player.lua]: Moved to data/stages.lua and created /reload stages

**Review by mattyx14 — APPROVED** [review]
marge to  lasted changed are good very nice work

**Comment by dudantas** [comment]
> marge to lasted changed are good very nice work

Thanks, we still need to correct the speed of the monsters, but soon it will be ready and I will go up so we can merge.

**Review by gpedro — APPROVED** [review]
LGTM;

**Comment by dudantas** [comment]
Speed in the XML of the monsters fixed.

**Comment by worthdavi** [comment]
Where is the credits for skillstages creator? :P @LukSrT

**Comment by slawkens** [comment]
To be honest, I don't like that stages are saved in Lua instead of XML.

This will make it harder to parse it by AAC makers.

Why no keep the format of the original stages.xml in TFS as a standard?

**Pull request merged**